src/provider/default.ts

```diff
@@ -159,7 +159,12 @@ export class Provider implements ProviderInterface {
       body: stringify(request),
       headers,
     })
-      .then((res) => res.text())
+      .then((res) => {
+        if (res.status !== 200) {
```

Comment:
sure it's always 200 and never 201 POST?

```suggestion
        if (res.status >= 400) {
```